### PR TITLE
[feat] JupyterHub integration

### PR DESCRIPTION
**Pull request by maxkochubey**
* Dockerfile for multiple types of Jupyter notebooks added
* Makefile refactored
* Docker build turned off in travis yaml
* JupyterLab helm chart removed
* JupyterLab container files removed
* Internal location for dev packages added to Dockerfile
* Helm vars removed
* Updates for Jenkins pipeline

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -38,6 +38,6 @@ matrix:
     #####################
     ### DOCKER IMAGES ###
     #####################
-    - language: bash
-      script:
-        - make docker-build-jupyterlab
\ No newline at end of file
+    # - language: bash
```
Comment (mcdoker18) [inline .travis.yml]: Remove these lines

File: scripts/Jenkinsfile [hunk 2]
```diff
@@ -137,19 +119,47 @@ pipeline {
...
+            steps {
+                script {
+                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
+                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
+                                    """
+                    cicdLibrary.buildDockerImage(
+                        'base-notebook',
+                        '.',
+                        'containers/jupyter-stacks/Dockerfile',
+                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base".replaceAll("\\s+"," ")
```
Comment (mcdoker18) [inline scripts/Jenkinsfile]: What is a reason to use the `replaceAll` function?

File: scripts/Jenkinsfile [hunk 3]
```diff
@@ -137,19 +119,47 @@ pipeline {
...
+                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow".replaceAll("\\s+"," ")
+                    )
                 }
-                stage("Upload Jupyterlab Docker image") {
-                    steps {
-                        script {
-                            cicdLibrary.uploadDockerImage('odahu-flow-jupyterlab', env.param_stable_release.toBoolean() && env.param_dockerhub_publishing_enabled.toBoolean())
-                        }
-                    }
+                script {
```
Comment (mcdoker18) [inline scripts/Jenkinsfile]: Please, use an another stage for images pushing

File: scripts/Jenkinsfile (hunk 2 above)
Comment (maxkochubey) [reply scripts/Jenkinsfile]: Jenkins job was unable to pass multiline string as single variable, so job was failing with
```
"docker build" requires exactly 1 argument.
```
(see build 7 of `Build_OdahuFlowJupyterlabPlugin_Test_JHub` job)

File: .travis.yml (hunk 1 above)
Comment (maxkochubey) [reply .travis.yml]: Removed in ccc1c8d0e41b4c4efd4f58c14887b2bde0e51a27.

File: scripts/Jenkinsfile (hunk 2 above)
Comment (mcdoker18) [reply scripts/Jenkinsfile]:
```groovy
                 script {
                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
                                    """
                    cicdLibrary.buildDockerImage(
                        'base-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base".replaceAll("\\s+"," ")
                    )
                    cicdLibrary.buildDockerImage(
                        'datascience-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=datascience".replaceAll("\\s+"," ")
                    )
                    cicdLibrary.buildDockerImage(
                        'tensorflow-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow".replaceAll("\\s+"," ")
                    )
                }
```
`->`
```groovy
                 script {
                    def buildArgs = """--build-arg NPM_EXTRA_INDEX_URL=https://${env.param_npm_private_url}
                                       --build-arg ODAHU_PLUGIN_VERSION=${Globals.buildVersion}
                                    """.replaceAll("\\s+"," ")
                    cicdLibrary.buildDockerImage(
                        'base-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=base"
                    )
                    cicdLibrary.buildDockerImage(
                        'datascience-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=datascience"
                    )
                    cicdLibrary.buildDockerImage(
                        'tensorflow-notebook',
                        '.',
                        'containers/jupyter-stacks/Dockerfile',
                        "${buildArgs} --build-arg NOTEBOOK_TYPE=tensorflow"
                    )
                }
```

Comment (maxkochubey) [reply scripts/Jenkinsfile]: yup, already did in 2f1e4dc38b894d110c67c4852b69c6664b4d1955

**Pull request merged**